File: QuickMobileTest/QuickMobileTest/Data/Resource List/Entity/AddressResponse.swift
//
//  ResourceAddress.swift
//  QuickMobileTest
//
//  Created by Balaji Galave on 5/18/19.
//

import Foundation

struct AddressResponse : Codable {
    let address1: String?
    let label: String?
    let zipCode: String?
    let city: String?
    let state: String?
    let country: String?
    let gps: Gps?
}

struct Gps: Codable {
    let latitude: String?
    let longitude: String?
}

File: app/src/main/java/com/quickseries/mobiletest/data/resources/AddressResponse.kt
package com.quickseries.mobiletest.data.resources

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Response item containing informations about an address.
 */
@JsonClass(generateAdapter = true)
data class AddressResponse(
    @Json(name = "address1")
    val address1: String? = null,
    @Json(name = "label")
    val label: String? = null,
    @Json(name = "zipCode")
    val zipCode: String? = null,
    @Json(name = "city")
    val city: String? = null,
    @Json(name = "state")
    val state: String? = null,
    @Json(name = "country")
    val country: String? = null,
    @Json(name = "gps")
    val gps: GpsResponse = GpsResponse()
)